ResearchKit/Common/ORKBaseGraphView.h

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (c) 2015, Apple Inc. All rights reserved.
+ 
+ Redistribution and use in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ 
+ 1.  Redistributions of source code must retain the above copyright notice, this
+ list of conditions and the following disclaimer.
+ 
+ 2.  Redistributions in binary form must reproduce the above copyright notice, 
+ this list of conditions and the following disclaimer in the documentation and/or 
+ other materials provided with the distribution. 
+ 
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors 
+ may be used to endorse or promote products derived from this software without 
+ specific prior written permission. No license is granted to the trademarks of 
+ the copyright holders even if such marks are included in this software. 
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE 
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL 
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, 
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+
+ 
+#import <UIKit/UIKit.h>
+#import <ResearchKit/ORKDefines.h>
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol ORKBaseGraphViewDelegate;
+
+/**
+ *  IMPORTANT: THIS IS AN ABSTRACT CLASS. IT HOLDS PROPERTIES & METHODS COMMON TO CLASSES LIKE ORKLineGraphView & ORKDiscreteGraphView.
+ */
+ORK_CLASS_AVAILABLE
+@interface ORKBaseGraphView : UIView
+
+@property (nonatomic, readonly) CGFloat minimumValue;
+
+@property (nonatomic, readonly) CGFloat maximumValue;
+
+@property (nonatomic, getter=isLandscapeMode) BOOL landscapeMode;
+
+@property (nonatomic) BOOL showsVerticalReferenceLines;
+
+@property (nonatomic, weak, nullable) id <ORKBaseGraphViewDelegate> delegate;
+
+/* Appearance */
+
+@property (nonatomic, strong, nullable) UIColor *tintColor;
+
+@property (nonatomic, strong, nullable) UIColor *axisColor;
+
+@property (nonatomic, strong, nullable) UIColor *axisTitleColor;
+
+@property (nonatomic, strong, nullable) UIFont *axisTitleFont;
+
+@property (nonatomic, strong, nullable) UIColor *referenceLineColor;
+
+@property (nonatomic, strong, nullable) UIColor *scrubberThumbColor;
+
+@property (nonatomic, strong, nullable) UIColor *scrubberLineColor;
+
+@property (nonatomic, strong, nullable) UIPanGestureRecognizer *panGestureRecognizer;
+
+@property (nonatomic, strong, nullable) NSString *emptyText;
```

Comment:
I think the configuration is a useful feature, but it might make sense to factor it out into a style object and provide pre-defined styles.

`tintColor` doesn't need to be re-defined here, it lives on `UIView`